R/verb-select.R

```diff
@@ -166,6 +167,22 @@ rename_groups <- function(lazy_query, vars) {
   lazy_query
 }
 
+rename_order <- function(lazy_query, vars) {
+  old2new <- set_names(names(vars), vars)
+  order <- op_sort(lazy_query)
+
+  desc <- purrr::map_lgl(order, ~ is_call(.x, "desc", n = 1L))
+  order <- purrr::map_if(order, desc, ~ call_args(.x)[[1L]])
+  order <- purrr::map_chr(order, as_name)
```

Comment:
I don't quite follow. `check_window_order_dots()` ensures that each element of `order` is either a symbol or `desc(<symbol>)`.  And these two steps extract the underlying symbol and convert it to a string so that it can be mapped more easily.

Reply:
When is an element of `order` not already going to be a name?

Reply:
After line 175 every element of `order` is a symbol. Then `as_name()` converts the symbols to strings.

Reply:
I just looked into the documentation

> `rlang::as_name()` is the _opposite_ of `base::as.name()`

which is indeed quite mean 😄